core/devmode/src/main/java/io/quarkus/dev/JavaCompilationProvider.java

```diff
@@ -48,9 +49,21 @@ public void compile(Set<File> filesToCompile, Context context) {
             fileManager.setLocation(StandardLocation.CLASS_PATH, context.getClasspath());
             fileManager.setLocation(StandardLocation.CLASS_OUTPUT, Collections.singleton(context.getOutputDirectory()));
 
+            List<String> compilerOptions = new ArrayList<String>();
```

Comment:
Shouldn't we make this a `Set` now that arbitrary options are coming in?

Reply:
The order of arguments might matter in some cases (`-source` would a good example were it not handled in a special way here: its argument has to come right after the option; it can also share the same argument as `-target`, so even an ordered set would break it).  I see what you mean, though.  Maybe it would make sense to do something smarter than a simple `addAll` of the fixed `COMPILER_OPTIONS`.

Reply:
How about a `LinkedHashSet` which preserved order?

Reply:
Still no good: `-target` `8` `-target` `9` yields `-target` `8` `9`.

Reply:
Great point. So we'll need some more sophisticated handling here.

Reply:
I put the more sophisticated flag handling in a separate class (`CompilerFlags`) and made a unit test for it.